src/librustc_codegen_llvm/back/write.rs

```diff
@@ -743,13 +742,13 @@ pub(crate) unsafe fn codegen(
             // We can't use the same module for asm and binary output, because that triggers
             // various errors like invalid IR or broken binaries, so we might have to clone the
             // module to produce the asm output
-            let llmod = if config.emit_obj { llvm::LLVMCloneModule(llmod) } else { llmod };
+            let llmod = if config_emit_obj_normal { llvm::LLVMCloneModule(llmod) } else { llmod };
             with_codegen(tm, llmod, config.no_builtins, |cpm| {
                 write_output_file(diag_handler, tm, cpm, llmod, &path, llvm::FileType::AssemblyFile)
             })?;
```

Comment:
Something like this may be best structured by extracting this to a helper function and doing something like:

```rust
if config.emit_asm {
    emit_asm_file(the, arguments);
}

if config_emit_obj_normal {
    if config.no_integrated_as {
        if !config.emit_asm {
            emit_asm_file(the, same, arguments);
        }
        // ...
    } else {
        // emit the object file via llvm
    }
}
```

Reply:
This won't be necessary if #70345 lands... so I will leave this alone for now. We can possibly re-visit later if #70345 doesn't land.